### PR TITLE
Handle errors other than Timeout in keyword 'Handle Alert'

### PR DESCRIPTION
**Pull request by Zeckie**
Handle TimeoutException separately in AlertKeywords._wait_alert
Raise the current exception only for TimeoutException and add new except block for WebDriverException.
https://github.com/robotframework/SeleniumLibrary/issues/1500

### CODE REVIEW

**Comment by aaltat** [comment]
I understand what you have done and I can see the reasoning behind. Only thing to do is to write some test for the chance. What you think, is this easy to test in the unit or in the acceptance tests

File: src/SeleniumLibrary/keywords/alert.py [hunk 1]
```diff
@@ -133,6 +133,9 @@ def _wait_alert(self, timeout=None):
...
         try:
             return wait.until(EC.alert_is_present())
-        except WebDriverException:
+        except TimeoutException:
             raise AssertionError('Alert not found in %s.'
-                                 % secs_to_timestr(timeout))
+                             % secs_to_timestr(timeout))
+        except WebDriverException as e:
+            raise AssertionError('An exception occurred waiting for alert: %s'
+                                 % str(e))
```
Comment (pekkaklarck) [inline src/SeleniumLibrary/keywords/alert.py]: `str(e)` fails if the error contains non-ASCII characters on Python 2. Good news is that `str()` can just be removed and it ought to work fine.

`e` alone perhaps looks a bit odd and could be renamed to `err` at the same time. I think that's used more in SL code base.

**Comment by aaltat** [comment]
It looks like something broken in the alert handling and test in CI fails. Could you take a look and fix the issue.

**Comment by aaltat** [comment]
In some of the environments there are constant failures:
```
FAIL: Acceptance.Keywords.Alerts.Handle Alert when popup window closes
Expected error 'STARTS:An exception occurred waiting for alert' but got 'Alert not found in 1 second
```
Unfortunately the log.html is not saved anywhere from the Travis runs.

I wonder is this a timing issue and would this be easier to test in the unit level.

**Comment by Zeckie** [comment]
> Unfortunately the log.html is not saved anywhere from the Travis runs.
> 
> I wonder is this a timing issue and would this be easier to test in the unit level.

Yes, it would be much easier to troubleshoot if we could see the log.html 
I agree that it looks like a timing issue - will try increasing the timeout & put a teardown so the new test failing won't cause other tests to fail.

**Comment by Zeckie** [comment]
oops, didn't mean to close

File: src/SeleniumLibrary/keywords/alert.py (hunk 1 above)
Comment (Zeckie) [reply src/SeleniumLibrary/keywords/alert.py]: I've made those changes in commit https://github.com/robotframework/SeleniumLibrary/pull/1501/commits/d61a8030d313bb2f697e07c4334e7992ba9570e3

**Comment by aaltat** [comment]
Fixed one minor thing and then this ready to go. Thank you for the contribution.

**Pull request merged**